account_move_default_journal/models/account_move.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def _search_default_journal(self, journal_types):
+        company_id = self._context.get("default_company_id", self.env.company.id)
+        domain = [
+            ("company_id", "=", company_id),
+            ("type", "in", journal_types),
+        ]
+        if self._context.get("default_currency_id"):
+            domain.append(("currency_id", "=", self._context["default_currency_id"]))
+
+        # Use the appropriate config parameter for the journal type
+        journal_id = None
+        if "sale" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.sale_journal_id"
+            )
+        elif "purchase" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.purchase_journal_id"
+            )
+        elif "general" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.general_journal_id"
+            )
+
+        # Try to get the journal, or default back to normal behavior
```

Comment:
If you use option 2 above (related field in res.config.settings to journal fields in company, you will have a journal you could directly return.

However if there is a default currency in the context that differs from the journal currency, you must still default to the standard search method.